Source/Dafny/Resolver/RunAllTestsMainMethod.cs

```diff
@@ -10,6 +10,10 @@ public class RunAllTestsMainMethod : IRewriter {
   public RunAllTestsMainMethod(ErrorReporter reporter) : base(reporter) {
   }
 
+  /// <summary>
+  /// Verifies that there isn't already a main method, and then creates
+  /// one with no body (to be filled in by PostResolve()).
```

Comment:
Just out of curiosity, why does this already need to happen in the PreResolve?

Reply:
I initially started with needing PreResolve because I thought I'd be generating unresolved AST nodes, but then realized I needed at least the test return types to be resolved. It might be feasible to move creating the main method to PostResolve as well, but then I'd have to create a resolved method so it might be more significant refactoring (I tried just moving the existing code and hit errors). Given the `PreResolve(Program)` seems like a good hook to add for rewriters in general, I'd prefer to leave this as is now that it's working.